pinot-broker/src/main/java/org/apache/pinot/broker/querylog/QueryLogger.java

```diff
@@ -255,6 +257,12 @@ void doFormat(StringBuilder builder, QueryLogger logger, QueryLogParams params)
           builder.append(CommonConstants.UNKNOWN);
         }
       }
+    },
+    IS_MULTI_STAGE_QUERY_ENGINE("isMultiStageQueryEngine") {
```

Comment:
Alternatively, we could call this `queryEngine` with possible options being `singleStage` / `multiStage` so that it is open to future extensions.

Reply:
I like that idea